### PR TITLE
Spring cleansing from 3 modules

### LINKED ISSUE #894: [IntelliJ Plugin] Remove Spring compile time traces in modules cloudslang-compiler, cloudslang-entities, cloudslang-spi
The motivation for this is to maintain a single code base for IntelliJ Plugin and the code in the official cloud-slang repo.
The ultimate goal is to remove Spring dependencies (spring-core, spring-context, spring-beans, spring-aop etc) from the IntelliJ Plugin. Other refactoring solutions of the module should be considered if possible.

### PR DESCRIPTION
**Pull request by lucian-cm**
Fixes #894
'Removing' Spring to have 1 codebase between plugin and cloud slang modules
Signed-off-by: Lucian Musca <lucian-cristian.musca@hpe.com>

### CODE REVIEW

File: cloudslang-entities/src/main/java/configuration/SlangEntitiesSpringConfig.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+public class SlangEntitiesSpringConfig implements ApplicationContextAware {
+
+    private static ApplicationContext applicationContext;
+
+    public static Encryption[] getEncryptors() {
+        if (applicationContext != null) {
+            Map<String, Encryption> encryptorMap = applicationContext.getBeansOfType(Encryption.class);
+            return encryptorMap.values().toArray(new Encryption[encryptorMap.size()]);
+        } else {
+            throw new RuntimeException("ApplicationContextProvider bean missing");
```
Comment (Bonczidai) [inline cloudslang-entities/src/main/java/configuration/SlangEntitiesSpringConfig.java]: IMHO in this case we should use `ApplicationContext bean missing` message - and extract it as const (it also used in `EncryptionProvider`)

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/encryption/EncryptionProvider.java [hunk 2]
```diff
@@ -34,21 +34,25 @@ public static Encryption get() {
...
-                return encryptors[0];
-            } else if (encryptors.length == 2) {
-                return encryptors[0] instanceof DummyEncryptor ? encryptors[1] : encryptors[0];
-            } else {
-                throw new RuntimeException("Too many (" + encryptors.length + ") encryptors found");
+        Encryption[] encryptors;
+        try {
+            encryptors = SlangEntitiesSpringConfig.getEncryptors();
+        } catch (Exception theGivenEx) {
+            if (!StringUtils.containsIgnoreCase(theGivenEx.getMessage(), "ApplicationContextProvider bean missing")) {
```
Comment (Bonczidai) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/encryption/EncryptionProvider.java]: hmm - do we need the negation here?

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java [hunk 3]
```diff
@@ -106,7 +95,6 @@
             asList(ScoreLangConstants.LOOP_KEY, SlangTextualKeys.DO_KEY, SlangTextualKeys.NAVIGATION_KEY);
     private List<String> parallelLoopValidKeywords = asList(SlangTextualKeys.DO_KEY, SlangTextualKeys.FOR_KEY);
 
-    @PostConstruct
```
Comment (Bonczidai) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java]: maybe we should leave this as a comment to mark the logic or switch to a builder / constructor approach to call this method in the constructor after all the setters - WDYT? Is it good enough to have the init logic only in spring config class?

File: cloudslang-entities/src/main/java/configuration/SlangEntitiesSpringConfig.java (hunk 1 above)
Comment (lucian-cm) [inline cloudslang-entities/src/main/java/configuration/SlangEntitiesSpringConfig.java]: Nice catch 👍  Please see last commit.

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/encryption/EncryptionProvider.java (hunk 2 above)
Comment (lucian-cm) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/encryption/EncryptionProvider.java]: I don't like negations as well, but in this case it is expressing the intention of the code 👍

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java (hunk 3 above)
Comment (lucian-cm) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java]: Added comment to mark it as post construction method

**Pull request merged**